pkg/controller/egress_router/egress_router_controller.go

```diff
@@ -0,0 +1,211 @@
+			log.Printf("Egress Router %s seems to have been deleted\n", request.NamespacedName)
+			return reconcile.Result{}, nil
+		}
+		log.Println(err)
+		return reconcile.Result{}, err
+	}
+
+	// Check to see if we already know this object
+	existing := r.egressrouters[request.NamespacedName]
+	if existing != nil {
+		if !reflect.DeepEqual(obj.Spec, existing.spec) {
+			log.Printf("Egress Router %s has changed, refreshing\n", request.NamespacedName)
+			delete(r.egressrouters, request.NamespacedName)
+			existing = nil
+		}
+	}
+
+	if existing == nil {
+		log.Printf("Creating a new Egress Router")
+		// Set owner reference to the controller
+		boolTrue := bool(true)
+		EgressRouterOwnerReferences := []v1.OwnerReference{
+			{
+				APIVersion: "network.operator.openshift.io/v1",
+				Kind:       "EgressRouter",
+				Name:       obj.Name,
+				UID:        obj.UID,
+				Controller: &boolTrue,
+			},
+		}
+		objs, err := ensureEgressRouter(manifestDir, request.Namespace, obj)
+		for _, obj := range objs {
+			log.Printf("Assigning owner references")
+			obj.SetOwnerReferences(EgressRouterOwnerReferences)
+			log.Printf("Applying manifest")
+			if err := apply.ApplyObject(context.TODO(), r.client, obj); err != nil {
+				log.Printf("could not apply operator config: %v", err)
+				r.status.SetDegraded(statusmanager.EgressRouterConfig, "UpdateEgressRouterConfig",
+					fmt.Sprintf("Error while reconciling EgressRouter: %v", err))
+			}
+		}
+
+		if err != nil {
+			log.Println(err)
+			r.egressrouterErrs[request.NamespacedName] =
+				errors.Wrapf(err, "could not reconcile Egress Router %s", request.NamespacedName)
+			r.setStatus()
+			return reconcile.Result{}, err
+		}
+		r.egressrouters[request.NamespacedName] = existing
+	}
+
+	if err != nil {
+		log.Println(err)
+		r.egressrouterErrs[request.NamespacedName] =
+			errors.Wrapf(err, "could not reconcile Egress Router %s", request.NamespacedName)
+		r.setStatus()
+		return reconcile.Result{}, err
+	}
+
+	log.Println("successful reconciliation")
+	delete(r.egressrouterErrs, request.NamespacedName)
+	r.setStatus()
+	return reconcile.Result{RequeueAfter: ResyncPeriod}, nil
+}
+
+// setStatus summarizes the status of all Egress Router objects and updates the statusmanager
+// as appropriate.
+func (r *EgressRouterReconciler) setStatus() {
+	if len(r.egressrouterErrs) == 0 {
+		r.status.SetNotDegraded(statusmanager.EgressRouterConfig)
+	} else {
+		msgs := []string{}
+		for _, e := range r.egressrouterErrs {
+			msgs = append(msgs, e.Error())
+		}
+		r.status.SetDegraded(statusmanager.EgressRouterConfig, "EgressRouterError", strings.Join(msgs, ", "))
+	}
+}
+
+func isItValidCidr(cidr string) bool {
+	_, _, err := net.ParseCIDR(cidr)
+	if err != nil {
+		log.Println(err)
+		return false
+	}
+	return true
+}
+
+func isItValidIPAddress(ip string) bool {
+	return net.ParseIP(ip) != nil
+}
+
+func ensureEgressRouter(manifestDir string, namespace string, router *netopv1.EgressRouter) ([]*uns.Unstructured, error) {
```

Comment:
suggestion: ensureEgressRouter should also apply the object to the apisever. That makes error handling in the controller simpler.

Reply:
thus, it would probably be easier to write this if it were a method on the controller.